pkg/option/option.go

```diff
@@ -181,27 +180,19 @@ func (opt *Options) renameLegacyClusterRoles() {
 	fmtLogger := fmt.Printf // Importing logger here is an import cycle, so use fmt instead.
 	if opt.ClusterRole == "writer" {
 		opt.ClusterRole = "primary"
-		fmtLogger(warning, "writer", "primary")
+		_, _ = fmtLogger(warning, "writer", "primary")
 	}
 	if opt.ClusterRole == "reader" {
 		opt.ClusterRole = "secondary"
-		fmtLogger(warning, "reader", "secondary")
+		_, _ = fmtLogger(warning, "reader", "secondary")
 	}
 }
 
 // Parse parses all arguments, returns normal message without error if --help/--version set.
-func (opt *Options) Parse() (string, error) {
+func (opt *Options) Parse() error {
 	err := opt.flags.Parse(os.Args[1:])
 	if err != nil {
-		return "", err
-	}
-
-	if opt.ShowVersion {
-		return version.Short, nil
-	}
-
-	if opt.ShowHelp {
-		return opt.flags.FlagUsages(), nil
```

Comment:
Removing these lines changes the logic, that it may fail to show the version or help information if the rest code returns an error.

Reply:
hi @localvar, I think it's okay for the program to return an error if the user includes some incorrect commands when starting the service while checking the version or help information.

Reply:
I still think it's okay. Consider the following examples, 

```
# Same behavior
easegress-server --help
easegress-server some-not-exists-command --help

# Different behaviors
// Previously, help information was displayed, but now an error is returned instead.
easegress-server -f not_exists_file.yaml --help  
```

where only an incorrectly specified configuration file could cause different behavior between the two. 

If the user specifies an incorrect configuration file (or if the file content is incorrect), returning an error directly without showing help may is acceptable.

Reply:
No, if the user wants the version and usage, we need always show the information, we all know there are jokes that a user asks where the `any` key is, so we need to keep the help information as easier to get as possible.

and from the current code, not only an incorrect configuration file will return an error.

and from the maintenance view, the code will change from time to time, even if only an incorrect configuration file returns an error now, developers may add other error cases later, and it is very possible for them to update these codes at the same time, that's result in a bad user experience.

Reply:
Do you have any good suggestions, or should we just close it?

Reply:
```go
	if opt.ShowVersion || opt.ShowHelp {
		return nil
	}
```